Remove unnecessary python 3.6 conditionals

jedcunningham opened the pull request:
Since Python 3.7 is now the lowest supported version, we no longer need
to have conditionals to support 3.6.

potiuk reviewed (changes requested):
I think we should only remove it after we release 2.3.0. There was already a case when pyupgrde automatically removed if for 3.6 but the intention for that PR was to chgerry-pick to 2.2.4:

https://github.com/apache/airflow/pull/19608#issuecomment-1002229080

So until we intend to cherry-pick more than just occassinal bugfixes to 2.2.* I think we should keep the "if 3.6" everywhere.

potiuk commented on Dockerfile.ci:
```diff
@@ -25,7 +25,7 @@ ARG AIRFLOW_VERSION="2.2.0.dev0"
 ARG AIRFLOW_IMAGE_REPOSITORY="https://github.com/apache/airflow"
 
 # By increasing this number we can do force build of all dependencies
-ARG DEPENDENCIES_EPOCH_NUMBER="6"
```
That was accidental I think :)

jedcunningham replied:
This was bumped because `pep562` was removed.

potiuk replied:
Ah I see :).

uranusjr replied:
Actually since `pep562` is only installed on `python_version < '3.7'`, I don’t think we even need to rebuild the dependencies because it is never actually installed in any of the images right now.

jedcunningham replied:
Good point, removing this.

potiuk commented on setup.cfg:
```diff
@@ -131,7 +131,6 @@ install_requires =
     openapi-spec-validator>=0.2.4
     packaging>=14.0
     pendulum~=2.0
-    pep562~=1.0;python_version<"3.7"
```
This one we can likely remove even now.

jedcunningham commented on airflow/__init__.py:
```diff
@@ -47,8 +47,6 @@
 login: Optional[Callable] = None
 
-PY36 = sys.version_info >= (3, 6)
-PY37 = sys.version_info >= (3, 7)
```
I'm actually going to leave these here and in __all__ so we don't break backcompat.

jedcunningham committed:
Remove unnecessary python 3.6 conditionals

Since Python 3.7 is now the lowest supported version, we no longer need
to have conditionals to support 3.6.
